utils/event-publisher/src/event_handler.rs

```diff
@@ -0,0 +1,22 @@
+use crate::{
+    common::constants::MAX_BUFFER_MSGS, event_layer::EventLayer, publisher::MbusPublisher,
+};
+use mbus_api::event::EventMessage;
+
+/// Event handle.
+pub struct EventHandle {}
+
+impl EventHandle {
+    /// Initilize the buffer, starts the event publisher and return the layer for handling the event
+    /// traces.
+    pub fn init(mbus_url: String) -> EventLayer {
+        let (send, recv) = tokio::sync::mpsc::channel::<EventMessage>(MAX_BUFFER_MSGS);
+
+        let layer = EventLayer::new(send);
+        tokio::spawn(async move {
+            let publisher = MbusPublisher::new(&mbus_url).await;
+            publisher.publish_events(recv).await;
```

Comment:
missing shutdown handler, or is that handled by the publisher already?

Reply:
Currently I am just printing out the publish errors without closing the receiver. 

If the receiver is disconnected due to some error, the channel should be recreated and the sender handle in event layer should also be replaced with the sender from the recreated one. Should I handle the disconnection this way?

Reply:
I mean shutdown as in cint or sigterm being received? 
We have a utils shutdown crate which let's us select futures with shutdown so we can exit gracefully, and for example flush any remaining events, but if we don't need to flush anything as long as this doesn't block the process from exiting then it's fine.

Reply:
And it should be able to fail without returning a non-zero error code, because the bus might be unreachable (e.g. helm chart uninstall).

Reply:
Even if the nats connection is lost or nats is uninstalled, the thread will continue to try to connect to nats.